pkg/apis/sriovnetwork/v1/helper.go

```diff
@@ -84,9 +84,12 @@ func (p *SriovNetworkNodePolicy) Apply(state *SriovNetworkNodeState) {
 			log.Info("Update interface", "name:", iface.Name)
 			interfaces = append(interfaces, Interface{
 				PciAddress: iface.PciAddress,
-				Mtu:        p.Spec.Mtu,
-				NumVfs:     p.Spec.NumVfs,
-				DeviceType: p.Spec.DeviceType,
+				Mtu:         p.Spec.Mtu,
+				NumVfs:      p.Spec.NumVfs,
+				TotalVfs:    p.Spec.TotalVfs,
+				LinkType:    p.Spec.LinkType,
+				EnableSriov: p.Spec.EnableSriov,
```

Comment:
you don't need the TotalVfs LinkType EnableSriov. we just need to make sure that if you have NumVfs is should be equal to the total number of VF.  EnableSriov should be set to true when you have NumVfs >0.   regarding the LinkType we can now assume it only ethernet